### PR TITLE
In trace sampler first PR

### PR DESCRIPTION
**Pull request by blumamir**
**Description:** 
This PR is based on #17209 
I was asked to split it into 3 parts as described in [CONTRIBUTING.md](https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#when-adding-a-new-component). So this it the first PR that includes:

> First PR should include the overall structure of the new component:
> Readme, configuration, and factory implementation usually using the helper factory structs.
> This PR is usually trivial to review, so the size limit does not apply to it.
> 

**Link to tracking Issue:**
[#17086](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17086)

**Testing:**
Added test similar to what I saw in other processors. Since this PR has the structure and not yet the implementation, future tests will be added in the next PR to test the processor logic

**Documentation:**
Added README.md to the new component + comments in code like I seen in other processors

### CODE REVIEW

**Review by atoulme — APPROVED** [review]
LGTM. Ideally the status should be “in development” but I think it’s fine to get this in and correct after.

File: processor/intracesampler/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+| Supported pipeline types | traces                        |
+| Distributions            | [contrib]                     |
+
+Supported pipeline types: traces
+
+The in-trace sampler can sample (remove) individual spans from a complete trace when they are too costly / verbose / noisy.
+
+To function properly, the sampler must receive "complete" traces from which it can derive contextual span properties used for sampling decisions, such as roots and leaves. It means that all spans for a given trace MUST be received by the same collector instance and that in-trace sampler processor MUST be placed in the trace pipeline after a processor that group spans by trace id (such as `tail_sampling` or `groupbytrace`)
+
+Just like any other sampler, the in-trace sampler can be configured such that for each span it can either sample (keeps all spans in a trace and does not change it) or not-sample (remove some of the spans in the trace). The `sampling_percentage` config option is used to control how many traces are passed as is and how many are sampled, so the user can save examples of both.
```
Comment (MovieStoreGuy) [inline processor/intracesampler/README.md:15]: It would be nice to highlight how this is different to [`filter processor`](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/filterprocessor) because I suspect that will be something that is asked.

Comment (MovieStoreGuy) [reply processor/intracesampler/README.md:15]: Perhaps some suggestions when this should be used instead of the filter processor.

**Review by MovieStoreGuy — APPROVED** [review]

**Comment by MovieStoreGuy** [comment]
Thank you for taking the time to split the PR up, it does make it easier to review and scope the feedback into actionable requests.

File: processor/intracesampler/README.md (hunk 1 above)
Comment (blumamir) [reply processor/intracesampler/README.md:15]: Thanks! great idea.

I added the following section to the README.md:

## Relation to Other Samplers
- [probabilisticsamplerprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/probabilisticsamplerprocessor) will either keep the entire trace or drop the entire trace. It is not able to sample out just parts of a trace.
- [tailsamplingprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/tailsamplingprocessor) will either keep the entire trace or drop the entire trace. It is not able to sample out just parts of a trace.
- [filterprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/filterprocessor) can drop individual spans and keep others, but it is not aware of the position of the span in the trace and thus can remove non-leaf spans and break the trace into multiple fragments. Also, it is not a sampler and will always filter out matching spans without allowing any samples to be kept.

It is recommended to use the intracesampler for traces that are valuable to keep, but includes spans which are not.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -1097,8 +1102,3 @@ updates:
     schedule:
       interval: "weekly"
       day: "wednesday"
-  - package-ecosystem: "gomod"
```
Comment (codeboten) [inline .github/dependabot.yml:1100]: This change seems unrelated

File: processor/intracesampler/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# In-Trace Sampling Processor
+
```
Comment (codeboten) [inline processor/intracesampler/README.md:2]: Doesn't need to be in this pr, but the status table can now be auto-generated, you can look at the haproxy files in [this change](https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/7ab7c53d32806e3986f353eee7e659349ac693a6) to see how it's done

**Comment by mx-psi** [comment]
It seems like there are a couple of unaddressed concerns on the comments above. I am removing the `ready to merge` label until these are addressed

File: processor/intracesampler/README.md (hunk 3 above)
Comment (atoulme) [reply processor/intracesampler/README.md:2]: mind if we leave this for the next PR? I'll take care of it.

File: .github/dependabot.yml (hunk 2 above)
Comment (atoulme) [reply .github/dependabot.yml:1100]: indeed please revert that chunk and I think we'll be ok.

Comment (atoulme) [reply .github/dependabot.yml:1100]: actually it's probably not unrelated. We have maxed out our dependabot updates, and this might be removed to make room for this new component (we go by alphabetical order).

Comment (atoulme) [reply .github/dependabot.yml]: Closing this thread as answered.

File: processor/intracesampler/README.md (hunk 3 above)
Comment (atoulme) [reply processor/intracesampler/README.md:2]: Alright, I think you said yes. Closing this thread.

**Comment by atoulme** [comment]
You're going to need one more rebase and `make gendependabot` to get back on top. Please do so after the release, you will also need to update all dependencies to 0.76.0. Sorry about that :/

**Comment by charlywazzap** [comment]
is this fully inactive?
